android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -82,6 +85,30 @@ private boolean needApiKey() {
 
   }
 
+  private static class DiscoveryServiceConnection implements ServiceConnection {
```

Comment:
Could you call this something like ClearCacheDiscoveryServiceConnection to make it clear what happens when you connect with it.